Formula/glibc.rb

```diff
@@ -0,0 +1,176 @@
+    "Linux kernel #{MINIMUM_LINUX_KERNEL_VERSION} (or later)"
+  end
+end
+
+class Glibc < Formula
+  desc "GNU C Library"
+  homepage "https://www.gnu.org/software/libc/"
+  url "https://ftp.gnu.org/gnu/glibc/glibc-2.23.tar.gz"
+  sha256 "2bd08abb24811cda62e17e61e9972f091f02a697df550e2e44ddcfb2255269d2"
+  license all_of: ["GPL-2.0-or-later", "LGPL-2.1-or-later"]
+
+  depends_on "binutils" => :build
+  depends_on GawkRequirement => :build
+  depends_on "linux-headers" => :build
+  depends_on MakeRequirement => :build
+  depends_on SedRequirement => :build
+  depends_on BrewedGlibcNotOlderRequirement
+  depends_on :linux
+  depends_on LinuxKernelRequirement
+
+  # GCC 4.7 or later is required.
+  fails_with gcc: "4.6"
+
+  def install
+    # Fix Error: `loc1@GLIBC_2.2.5' can't be versioned to common symbol 'loc1'
+    # See https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=
+    # Fixed in glibc 2.24
+    inreplace "misc/regexp.c", /^(char \*loc[12s]);$/, "\\1 __attribute__ ((nocommon));"
+
+    # Setting RPATH breaks glibc.
+    %w[
+      LDFLAGS LD_LIBRARY_PATH LD_RUN_PATH LIBRARY_PATH
+      HOMEBREW_DYNAMIC_LINKER HOMEBREW_LIBRARY_PATHS HOMEBREW_RPATH_PATHS
+    ].each { |x| ENV.delete x }
+
+    # Use brewed ld.so.preload rather than the hotst's /etc/ld.so.preload
+    inreplace "elf/rtld.c", '= "/etc/ld.so.preload";', '= SYSCONFDIR "/ld.so.preload";'
+
+    mkdir "build" do
+      args = [
+        "--disable-debug",
+        "--disable-dependency-tracking",
+        "--disable-silent-rules",
+        "--prefix=#{prefix}",
+        "--enable-obsolete-rpc",
+        "--without-selinux",
+        "--with-binutils=#{Formula["binutils"].bin}",
+        "--with-headers=#{Formula["linux-headers"].include}",
+      ]
+      system "../configure", *args
+      system "make", "all"
+      system "make", "install"
+      prefix.install_symlink "lib" => "lib64"
+    end
+  end
+
+  def post_install
+    # Fix permissions
+    chmod 0755, [lib/"ld-#{version}.so", lib/"libc-#{version}.so"]
+
+    # Install ld.so symlink.
+    ln_sf lib/"ld-linux-x86-64.so.2", HOMEBREW_PREFIX/"lib/ld.so"
+
+    # Compile locale definition files
+    mkdir_p lib/"locale"
+    locales = ENV.map { |k, v| v if k[/^LANG$|^LC_/] && v != "C" && !v.start_with?("C.") }.compact
+    # en_US.UTF-8 is required by gawk make check
+    locales = (locales + ["en_US.UTF-8"]).sort.uniq
+    ohai "Installing locale data for #{locales.join(" ")}"
+    locales.each do |locale|
+      lang, charmap = locale.split(".", 2)
+      if charmap.present?
+        charmap = "UTF-8" if charmap == "utf8"
+        system bin/"localedef", "-i", lang, "-f", charmap, locale
+      else
+        system bin/"localedef", "-i", lang, locale
+      end
+    end
+
+    # Set the local time zone
+    sys_localtime = Pathname.new "/etc/localtime"
+    brew_localtime = Pathname.new prefix/"etc/localtime"
+    (prefix/"etc").install_symlink sys_localtime if sys_localtime.exist? && !brew_localtime.exist?
```

Comment:
What if this symlink exists but it's pointing to the wrong place?

Reply:
I guess that if `/etc/localtime` is missing or points to the wrong place, you might have bigger issues with your system.
https://www.freedesktop.org/software/systemd/man/localtime.html

It looks like a quite common file on unix. On ubuntu 16.04:
> lrwxrwxrwx 1 root root 27 May 25  2020 /etc/localtime -> /usr/share/zoneinfo/Etc/UTC